### ISSUE TITLE
expanding from ulogger-android - a simple text based collectible AR game feasible?

### ISSUE THREAD

**Issue by khlsvr** [bfabiszewski/ulogger-android#102]
Hey, I was not sure where I should write this. Maybe I should have e-mailed the creator of this tool, Bartek, or ask about this somwhere else, but then again maybe someone here could help me out. 

I have been thinking of this very simple AR game, and it all would fit well by just adding couple extra features to this app.

So what the game would do in the ulogger is that every time the ulogger logs another trackpoint, it would do a calculation and if certain conditions are met, it would log a collectible being found. 

The conditions would be something like:

- Logged position is further than 50m from the previously logged collectible or rather, the GPS coordinates fall into another "cell". The coordinates would be rounded so that multiple positions would fall under one position, so let's say 50x50m area would be considered one and same place.  

- At minimum x minutes have passed since a collectible was found by your app in a given cell. So you can find collectible in one place only once every x minutes. (Rounding of time value would be used here too) 

With only just above rules, we could have a simple real world collectible game, where you can't accumulate collectibles while you are stationary e.g. at home or work, and you can't get them too fast by crossing over from cell 1 to cell 2 and back repeatedly, as a new collectible would spawn to cell 1 only after x minutes has passed.

The collectibles could be anything, for example pokemon. Let's say we have 1000 different pokemon in the game as collectibles. If we allow the collectible to be anything at random, we could devise some simple math formula with variables for the time and the coordinates and do calculation based on those values to end up an integer between 1 and 1000 to get the pokemon in question. This way even while we are playing this game offline, me and my friend would be able to find a same collectible around the same time in the same place in the world.   

No graphics required. Just a textual log of what was found, when and where.

Is this something that could be worked on top of ulogger-android? Just some extra math function and show the results in the UI. I am very limited in skill in programming. I have never done anything with Android, so if this is something no one else is willing to do for me, can anyone point me to the right direction. Where should I start if I wanted to try do this? Thanks

**Comment by bfabiszewski** [comment]
If I were to do it I'd rather start from scratch instead of building on µlogger. You probably don't need background logging, option to choose individual location providers, position uploading to remote server etc. Your code could be more simple and without some related complexities and limitations. There are a lot of good Android tutorials on logging location and geofencing.
On the other hand µlogger could also be modified to your needs. After all it depends on your skills, time you have and what you feel would be easier or give you more fun.

**Comment by khlsvr** [comment]
Ok! I should maybe search those tutorials. On the other hand, I would probably still be running ulogger unless the other app would be coded to save gpx files from the daily goings aswell. In the planned game, I would probably be playing it by having the phone in the pocket while I'm out, same way as just logging gpx, so maybe the background logging is needed in this app too, if I understand what background logging means in this context. 

I have even considered starting from a linux script since I have a little bit of experience from those. So after I have been out and have saved the gpx file from ulogger, I would then run that file through a script, which would then generate my "sightings" from the location/time combinations I logged. This is almost all I want in the end product. In the end I would want to do that on the fly, so if I reach my phone while I am out and check the app, It would show me my recent sightings.

**Comment by khlsvr** [comment]
I wrote a small python script for termux to track my path to a gpx file using the termux-location binary. I tested running it alongside ulogger and found it using less battery and suiting better for my situation (constant logging when out from home or not stationary for long somewhere). It will be a good base for this game I had planned here if I ever find the motivation to it after all. I will close this issue now. I am a very average coder so I won't be showing off that python script unless someone wants it.
